crates/catalog/rest/src/catalog.rs

```diff
@@ -497,13 +516,56 @@ impl RestCatalog {
             client: config.try_create_rest_client()?,
             config,
         };
-
+        catalog.fetch_access_token().await?;
+        catalog.client = catalog.config.try_create_rest_client()?;
```

Comment:
The python implementation update config first, then recreate client? Is this required or we can relax? cc @Fokko @flyrain

Reply:
Sorry, I don't get your point. I know that we should recreate http client, what I mean is that is it required that `update_config` happens before `fetch_access_token`? Seems java does this in reverse order, so I guess it doesn't matter.

Reply:
> what I mean is that is it required that update_config happens before fetch_access_token?

`fetch_access_token` happens before `update_config` in my code, and also in the python code.

Some config API also need authentication, such as `https://api.tabular.io/ws/v1/config`